### PR TITLE
Implementation of `OtlpHttpLogExporter`

### PR DESCRIPTION
**Pull request by owent**
Related to #1024 

## Changes

+ Move shared codes of OTLP recordable into `OtlpRecordableUtils` .
+ Create a new target `OtlpHttpClient` and move shared codes of HTTP exportor into this target .
+ Add `OtlpHttpLogExporter` .

@ThomsonTan @lalitb Could you please review the codes? There should be no API break change by now. Unit tests are already added.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

File: exporters/otlp/src/otlp_http_client.cc [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+  // A string to store the response body
+  std::string body_ = "";
+
+  // Whether to print the results from the callback
+  bool console_debug_ = false;
+};
+
+static inline char HexEncode(unsigned char byte)
+{
+  if (byte >= 10)
```
Comment (ThomsonTan) [inline exporters/otlp/src/otlp_http_client.cc:241]: Assert on byte < 16?

**Review by ThomsonTan — APPROVED** [review]

**Comment by lalitb** [comment]
@owent will merge it once it is updated to main.

File: exporters/otlp/src/otlp_http_client.cc (hunk 1 above)
Comment (owent) [reply exporters/otlp/src/otlp_http_client.cc:253]: Thanks and fixed.

**Comment by owent** [comment]
> @owent will merge it once it is updated to main.

Done

**Pull request merged**